crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -431,9 +431,25 @@ impl<'this, TX: DbTxMut<'this> + DbTx<'this>> DatabaseProvider<'this, TX> {
             .map(|(id, tx)| (id, tx.into()))
             .collect::<Vec<(u64, TransactionSigned)>>();
 
-        let senders =
+        let mut senders =
             self.get_or_take::<tables::TxSenders, TAKE>(first_transaction..=last_transaction)?;
 
+        // Recover senders manually if not found in db
+        let start_index = senders.len();
+        let end_index = transactions.len();
+        let missing_senders = end_index - start_index;
```

Comment:
I think there's a problem with such approach if you have `TxSenders` missing in the middle of the range, and not in the end. It is totally possible, because we prune data starting from `TxNumber = 0` and onward to the last row.

`self.get_or_take` returns a vec of `KeyValue`, so we can first calculate the `TxNumber`s missing from the database, then recover the signers for them and insert in the resulting vec respecting the initial order.